auth/src/main/java/org/aerogear/mobile/auth/utils/UserIdentityParser.java

```diff
@@ -56,13 +61,15 @@ public UserIdentityParser(final OIDCCredentials credential,
      * Parse the users first name from the {@link #userIdentity users identity}
      *
      * @return user's first name
-     * @throws JSONException if the {@link #FIRST_NAME} property cannot be retrieved from the user
-     *         identity.
      */
-    public String parseFirstName() throws JSONException {
+    public String parseFirstName() {
         String firstName = "";
         if (userIdentity != null && userIdentity.has(FIRST_NAME)) {
-            firstName = userIdentity.getString(FIRST_NAME);
+            try {
+                firstName = userIdentity.getString(FIRST_NAME);
+            } catch (JSONException e) {
+                LOG.debug(TAG, "Failed to get user first name from user identity", e);
+            }
         }
         return firstName;
```

Comment:
This function can be replaced with a one liner:

```java
return userIdentity == null ? "" : userIdentity.optString(FIRST_NAME, "");
```

Reply:
Nice, I'll refactor these functions